### PR TITLE
fix the edit command and classify the current snippet fix the save asset command

### PR DESCRIPTION
**Pull request by BishoyHanyRaafat**
#70

### CODE REVIEW

**Comment by BishoyHanyRaafat** [comment]
@hal-8999-alpha

File: README.md
Comment (mark-at-pieces) [inline README.md:20]: this is cool, however would be nice to see the materials on here that we can also list IE

Assets
Models, XYZ ,, note: the supported materals that we can list right now.

Comment (mark-at-pieces) [reply README.md:20]: @BishoyHanyRaafat lets also double check that these changes are reflected in the help command to ensure consistency

File: README.md
Comment (mark-at-pieces) [inline README.md:25]: lets double check that these are added to the help command as well

File: README.md
Comment (mark-at-pieces) [inline README.md:146]: these are looking good can we add a test for these examples w/ the edit command

File: src/pieces/api/assets.py
Comment (mark-at-pieces) [inline src/pieces/api/assets.py:170]: might be nice to include the asset name here as well, in the print statement

File: README.md
Comment (caleb-at-pieces) [inline README.md:131]: I still have to try this out but I am unsure about the paradigm if the 'current' asset. This might need to live as its own issue but how can I set the 'current' asset, also is there a good way to check what the current asset is?

File: src/pieces/api/assets.py
Comment (mark-at-pieces) [inline src/pieces/api/assets.py:190]: this is looking good, however the value can also be on the original.file.string.raw as well, so lets double check that,

Also... I would not enable the ability to update images at this point in time,

you can filter images by looking at the original(format).classification

File: src/pieces/api/assets.py
Comment (caleb-at-pieces) [inline src/pieces/api/assets.py:168]: Please try this with an OCR snippet. TO create an OCR snippet, copy and paste a screenshot of code into the desktop app.

OCR snippets are handled differently in the case of editing + reclassifying than normal assets. We might also need to make this its separate issue, unless you want to layer it into this PR.

File: src/pieces/api/assets.py
Comment (caleb-at-pieces) [inline src/pieces/api/assets.py:177]: having 'r' is nice for verbosity but not required, as read is the default

File: src/pieces/api/assets.py
Comment (mark-at-pieces) [inline src/pieces/api/assets.py:198]: will want to ensure that we are updating the property here is in line with what the value is on as well, 

IF if the value is on original.fragment.string.raw then we update the original.fragment.string.raw

if the value is on original.file.string.raw then we update the original.file.string.raw

File: src/pieces/api/assets.py
Comment (caleb-at-pieces) [inline src/pieces/api/assets.py:189]: Again, this flow is going to be different if it is an OCR snippet. Can be its own issue unless you want to layer it into this PR.

File: README.md
Comment (BishoyHanyRaafat) [reply README.md:131]: You can set the current asset by using the open command this is jayson idea and I like it but if u have a better way in mind lemme know

File: src/pieces/api/assets.py
Comment (caleb-at-pieces) [inline src/pieces/api/assets.py:190]: the value can also be stored in original.file.string.raw I'm pretty sure

File: src/pieces/api/assets.py
Comment (mark-at-pieces) [inline src/pieces/api/assets.py:199]: we dont necessarily need transferable:true here, because we are not using this value after the fact, meaning that we are not cacheing this value, and we are not displaying it to the user so the transferable can be false

File: src/pieces/api/assets.py
Comment (BishoyHanyRaafat) [reply src/pieces/api/assets.py:168]: Okay thank u I did not know that there is a different way to classify the ocr snippet I will try it and tell lwt you know

File: src/pieces/commands/assets.py
Comment (mark-at-pieces) [inline src/pieces/commands/assets.py:92]: I would probably just adjust this name to be specific as well

**Review by caleb-at-pieces — APPROVED** [review]
looking solid there's just some future work that needs to be done with the OCR snippets

File: src/pieces/commands/assets.py
Comment (mark-at-pieces) [reply src/pieces/commands/assets.py:92]: ie edit name
edit classification

File: src/pieces/commands/assets.py
Comment (mark-at-pieces) [inline src/pieces/commands/assets.py:83]: might want to adjust this command to something like update Asset transferable, which will refer to the asset core value

**Comment by BishoyHanyRaafat** [comment]
Can I know what will bw differ if it is ocr and what should I do?

**Review by mark-at-pieces — CHANGES_REQUESTED** [review]
let some comments, looking solid, i would just ensure that we are reflecting the changes in the readme in the help command, I would add some tests for the edit functionality, and I would update a few of the names.
 
 Really great stuff here please ping me when everything has been resloved.

File: README.md
Comment (BishoyHanyRaafat) [reply README.md:146]: sure I will add add a test for the edit command. There is currently a test for the edit

File: src/pieces/api/assets.py
Comment (BishoyHanyRaafat) [reply src/pieces/api/assets.py:198]: okay added

File: src/pieces/commands/assets.py
Comment (BishoyHanyRaafat) [reply src/pieces/commands/assets.py]: sorry that was an old comment and I removed

**Comment by BishoyHanyRaafat** [comment]
Any other review that I am missing? @caleb-at-pieces @mark-at-pieces

File: src/pieces/api/assets.py
Comment (caleb-at-pieces) [inline src/pieces/api/assets.py:169]: wouldn't this mean that it fails for all assets that aren't images?

File: src/pieces/api/assets.py
Comment (caleb-at-pieces) [inline src/pieces/api/assets.py:197]: the else here should also check if there is an original.file.string.raw

File: src/pieces/api/assets.py
Comment (caleb-at-pieces) [inline src/pieces/api/assets.py:205]: once again this can potentially be oiginal.file.string.raw

**Review by caleb-at-pieces — CHANGES_REQUESTED** [review]
i think the if statement for the image assets is flipped also we should handle the case where the asset's content is stored in origina.format.file.string.raw

**Comment by BishoyHanyRaafat** [comment]
@caleb-at-pieces I got you this time I was confused a bit at the first glance thank you for clarifying

**Comment by caleb-at-pieces** [comment]
@BishoyHanyRaafat it looks like mark forgot to approve this one so we might need to wait until he is back to merge it.

File: src/pieces/api/assets.py
Comment (mark-at-pieces) [inline src/pieces/api/assets.py:200]: little bug here

original.fragment -> original.file

**Review by mark-at-pieces — CHANGES_REQUESTED** [review]
1 little bug then im gtg

**Comment by BishoyHanyRaafat** [comment]
@mark-at-pieces

**Review by mark-at-pieces — APPROVED** [review]

**Pull request merged**
